src/runtime/COOTensor.cpp

```diff
@@ -0,0 +1,264 @@
+    const size_t first_elem_offset = info()->offset_first_element_in_bytes();
+
+    std::memset(tensor->buffer() + first_elem_offset, 0, total_size);
+
+    if(nnz() == 0)
+    {
+        return tensor;
+    }
+
+    for(size_t i = 0; i < _indices.size(); ++i)
+    {
+        const Coordinates     &c = _indices[i];
+        const uint8_t *block_ptr = buffer() + i * dense_vol * element_size;
+
+        size_t final_offset = 0;
+        for(size_t d = 0; d < sparse_dim(); ++d)
+        {
+            final_offset += c[d] * dense_volume(d+1);
+        }
+        final_offset *= element_size;
+
+        for(size_t j = 0; j < dense_vol; ++j)
+        {
+            const void *value_ptr = block_ptr + j * element_size;
+            uint8_t     *base_ptr = tensor->buffer() + final_offset + j * element_size;
+
+            std::memcpy(base_ptr, value_ptr, element_size);
+        }
+    }
+
+    return tensor;
+}
+
+Coordinates COOTensor::get_coordinates(size_t nth) const
+{
+    ARM_COMPUTE_ERROR_ON_MSG(nth >= nnz(), "Invalid index");
+
+    return _indices[nth];
+}
+
+const uint8_t *COOTensor::get_value(Coordinates coords) const
+{
+    ARM_COMPUTE_ERROR_ON_MSG(coords.num_dimensions() != info()->num_dimensions(), "Invalid coordinate dimension");
+    for(size_t i = 0; i < coords.num_dimensions(); ++i)
+    {
+        ARM_COMPUTE_ERROR_ON_MSG(static_cast<size_t>(coords[i]) >= info()->tensor_shape()[i], "Invalid coordinates shape");
+    }
+
+    const uint8_t    *data = static_cast<const uint8_t *>(buffer());
+    const size_t dense_vol = dense_volume(sparse_dim());
+
+    for(size_t i = 0; i < _indices.size(); ++i)
+    {
+        const Coordinates &c = _indices[i];
+        bool           match = false;
+
+        for(size_t d = 0; d < coords.num_dimensions(); ++d)
+        {
+            if(c[d] == coords[d])
+            {
+                match = true;
+            }
+        }
+        if(match)
+        {
+            return data + i * dense_vol * info()->element_size();
+        }
+    }
+
+    // This coordinates point to a zero value
+    return nullptr;
+}
+
+void COOTensor::associate_memory_group(IMemoryGroup *memory_group)
+{
+    _allocator.set_associated_memory_group(memory_group);
+}
+
+#ifdef ARM_COMPUTE_ASSERTS_ENABLED
+void COOTensor::print(std::ostream &os) const
+{
+    const uint8_t *data = static_cast<const uint8_t *>(buffer());
+
+    if(_indices.empty())
+    {
+        os << "index: [] values: []" << std::endl;
+        return;
+    }
+
+    for(size_t i = 0; i < _indices.size(); ++i)
+    {
+        const Coordinates &coord = _indices[i];
+        os << "index: [";
+        for (size_t j = 0; j < coord.num_dimensions(); ++j)
+        {
+            os << coord[j];
+            if (j < coord.num_dimensions() - 1) os << ", ";
+        }
+        os << "]  values: ";
+        print_values(os, data, i, dense_volume(sparse_dim()));
+    }
+}
+#endif // ARM_COMPUTE_ASSERTS_ENABLED
```

Comment:
We already have a print method in https://github.com/ARM-software/ComputeLibrary/blob/main/src/core/ITensor.cpp

We should move this in `void ITensor::print(std::ostream &s, IOFormatInfo io_fmt) const` 

There you can ask